src/assisted_test_infra/test_infra/utils/utils.py

```diff
@@ -561,7 +561,8 @@ def update_hosts(client, cluster_id, libvirt_nodes, update_hostnames=False, upda
     if not update_roles:
         roles = None
 
-    client.update_hosts(cluster_id=cluster_id, hosts_with_roles=roles, hosts_names=hostnames)
```

Comment:
Why not replacing it in this PR? Why split it into 2 PRs?

Reply:
It just needed more work and testings. Handled in this PR: https://github.com/openshift/assisted-test-infra/pull/1525